packages/workflow/test/WorkflowDataProxy.test.ts

```diff
@@ -283,4 +306,61 @@ describe('WorkflowDataProxy', () => {
 			}
 		});
 	});
+
+	describe('Pinned data with manual execution', () => {
+		const fixture = loadFixture('pindata');
+		const proxy = getProxyFromFixture(fixture.workflow, null, 'NotPinnedSet1', 'manual');
+
+		test('$(PinnedSet).item.json', () => {
+			expect(proxy.$('PinnedSet').item.json).toEqual({ firstName: 'Joe', lastName: 'Smith' });
+		});
+
+		test('$(PinnedSet).item.json.firstName', () => {
+			expect(proxy.$('PinnedSet').item.json.firstName).toEqual('Joe');
```

Comment:
Nit - Prefer `toBe` for primitives or object refs, `toEqual` for object content. Same elsewhere.

```suggestion
			expect(proxy.$('PinnedSet').item.json.firstName).toBe('Joe');
```